### PR TITLE
dev-python/python-ethtool: EAPI7, fix Homepage and SRC_URI

### PR DESCRIPTION
**Pull request by mm1ke**
```diff
--- python-ethtool-0.11.ebuild  2018-02-11 16:07:54.471707880 +0100
+++ python-ethtool-0.11-r1.ebuild       2019-01-29 14:47:36.607377947 +0100
@@ -1,27 +1,26 @@
-# Copyright 1999-2016 Gentoo Foundation
+# Copyright 1999-2019 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=5
+EAPI=7
 
 PYTHON_COMPAT=( python2_7 )
 
 # TODO: install scripts and their man pages
 
-inherit eutils distutils-r1
+inherit distutils-r1
 
 DESCRIPTION="Ethernet settings python bindings"
-HOMEPAGE="https://fedorahosted.org/python-ethtool/"
-SRC_URI="https://fedorahosted.org/releases/p/y/${PN}/${P}.tar.bz2"
+HOMEPAGE="https://pypi.org/project/ethtool/
+       https://github.com/fedora-python/python-ethtool"
+SRC_URI="https://github.com/fedora-python/python-ethtool/archive/v${PV}.tar.gz -> ${P}.tar.gz"
 
 LICENSE="GPL-2"
 SLOT="0"
 KEYWORDS="~amd64 ~amd64"
-IUSE=""
 
 DEPEND="dev-libs/libnl:3"
 RDEPEND="${DEPEND}"
 
-src_prepare() {
-       epatch "${FILESDIR}/${P}_include-net-if.h-for-IFF_-macros.patch"
-       distutils-r1_src_prepare
-}
+PATCHES=(
+       "${FILESDIR}/${P}_include-net-if.h-for-IFF_-macros.patch"
+)
```

Hi,

This PR/BUG updates dev-python/python-ethtool for EAPI7 and fixes the HOMEPAGE and SRC_URI.
Please review.

Closes: https://bugs.gentoo.org/676784
Signed-off-by: Michael Mair-Keimberger <m.mairkeimberger@gmail.com>

### CODE REVIEW

**Comment by gyakovlev** [comment]
while issue was there before, i think it's missing setuptools dep. add it to BDEPEND, since it's eapi7.

**Comment by gyakovlev** [comment]
btw, there's new version available upstream which supports up to python 3.7, can you bump? )

**Comment by mm1ke** [comment]
@gyakovlev 
I've added the missing setuptools as BDEPEND.
I also made a version bump to 0.14 with python3 support. However, i think it still needs some adjustments since it would install two broken symlinks right now:
```
lrwxrwxrwx 1 ai 2000  31 Feb  2 21:19 pethtool -> ../lib/python-exec/python-exec2
lrwxrwxrwx 1 ai 2000  31 Feb  2 21:19 pifconfig -> ../lib/python-exec/python-exec2
```
However, both script are available for python2.7 and python3.6 ```/usr/lib/python-exec/python2.7/{pethtool,pifconfig``` and ```/usr/lib/python-exec/python3.6/{pethtool,pifconfig}```

**Comment by gyakovlev** [comment]
thanks! merged.

I also added tests, removed duplicate keywords and found a bug in tests and submitted patch upstream https://github.com/fedora-python/python-ethtool/pull/54
